### PR TITLE
Make ARC work on non-standard MySQL port

### LINKED ISSUE #17: ARC2 fails to connect to MySQL running on non-standard port
I've set up an instance of Scalar running here at the University of Washington. As it is a shared hosting environment, we need to run our instance of MySQL on a unique port. I added a port to `system/application/config/database.php` as described in the CodeIgniter documentation:
    $db['default']['port'] = 1234;
This works and lets us connect to the database from the CI application, but apparently not from ARC2. I am getting these error messages multiple times on every page I access:
```
A PHP Error was encountered

Severity: Warning

Message: mysqli_connect(): (HY000/2005): Unknown MySQL server host 'oursubdomain.u.washington.edu:1234' (2)

Filename: store/ARC2_Store.php

Line Number: 51
```

It appears arc is not properly passing the separate `$port` parameter to `mysqli_connect` as is required.

### PR DESCRIPTION
**Pull request by zploskey**
This fixes #17.

### CODE REVIEW

**Comment by craigdietrich** [comment]
Excellent, thanks for taking this on!

A quick question, I'm curious about the ($ci->db->port != '') this block:

    if ($ci->db->port != '') {
        $hostname_port = $ci->db->hostname;
        $hostname = strstr($hostname_port, ':', true);
    } else {
        $hostname = $ci->db->hostname;
    }

Is this to trap in case config/database.php is set up like this?,

    $db['default']['hostname'] = '127.0.0.1:1234';  
    $db['default']['port'] = '1234';    

If so, I'm curious why someone would put the port number in the hostname field if it's already defined in the port field?

Apologies if I'm missing something obvious!

**Comment by zploskey** [comment]
Apparently this is how CodeIgniter stores the hostname if a port is provided, since some database drivers will accept the port in the hostname string like "hostname:port". Mysqli doesn't accept a hostname in this format, though, and that just so happens to be what ARC uses.

If CodeIgniter accepts hostnames configured like `$db['default']['hostname'] = '127.0.0.1:1234';` then it's not documented. The documentation says to use a separate 'port' key.

Basically that code says if there was a port set in the configuration file, get the hostname without the ":<port>" at the end and pass those along to ARC, since that's what it expects.

**Comment by craigdietrich** [comment]
Ah, I see, great find on that!  

I wonder to what extent the same change should be made directly to the ARC2 repository.  We don't plan to update Scalar's ARC directory for a while, but down the road it might be useful to know that know customization is needed if we do pulldown a new version?

**Comment by zploskey** [comment]
This seems like a patch that might be accepted upstream, if it's still being maintained. If not, the changes to the ARC2 code were pretty minimal, and could be merged with an updated version without much hassle.
